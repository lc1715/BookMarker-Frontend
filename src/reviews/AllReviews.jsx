import TimedMessage from '../common/TimedMessage';
//Material UI
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import Divider from '@mui/material/Divider';
import CloseIcon from '@mui/icons-material/Close';


//show all reviews on a book
function AllReviews({ allReviews, setShowAllReviews }) {
    console.log('allReviews=', allReviews)
    if (!allReviews.length) {
        return (
            <div>
                <Box sx={{ mt: 1, mb: 2, display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
                    <TimedMessage setState={setShowAllReviews} type={'info'} message={'There are no reviews on this book'} />
                </Box>
            </div >
        )
    }

    return (
        <div>
            <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
                <Box component='h2'
                    sx={{ fontSize: { lg: '21.7px' }, marginTop: '30px', marginLeft: '25px', textDecoration: 'underline' }}>
                    All Reviews:
                </Box>
                <IconButton onClick={() => (setShowAllReviews(false))} sx={{ mr: { xs: 1.36, lg: 2.2 } }} size="large" color='warning'>
                    <CloseIcon fontSize="inherit" />
                </IconButton>
            </Box>

            {allReviews.toReversed().map((review) => (
                <div key={review.id} >
                    <Box sx={{ mx: 3, mt: 1, mb: 4 }}>
                        <Typography sx={{ fontSize: { xs: 19, lg: 17 }, wordBreak: "break-word" }}>
                            {review.comment}
                        </Typography>

                        <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', fontSize: { xs: 18, lg: 16.4 }, color: '#424242', mt: 2, mb: 1 }}>
                            {review.username} - {new Date(review.created_at).toLocaleDateString()}
                        </Box>

                        < Divider sx={{ mb: '30px' }} />
                    </Box>
                </div>
            ))}
        </div >
    )
};

export default AllReviews;
