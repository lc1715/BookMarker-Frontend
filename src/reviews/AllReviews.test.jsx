import { expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import AllReviews from './AllReviews';

const allReviews = [{
    id: 1,
    comment: 'comment1',
    created_at: 'date1',
    username: 'user1',
    user_id: 2,
    volume_id: 'vol1'
}]

it('renders the AllReviews component', () => {
    render(
        <AllReviews allReviews={allReviews} />
    )
})

it('matches with snapshot', async () => {
    const { asFragment } = render(
        <AllReviews allReviews={allReviews} />
    )

    expect(asFragment()).toMatchSnapshot();
})

it('has the correct reviews data', async () => {
    render(
        <AllReviews allReviews={allReviews} />
    )

    expect(screen.getByText('comment1')).toBeInTheDocument();
});
