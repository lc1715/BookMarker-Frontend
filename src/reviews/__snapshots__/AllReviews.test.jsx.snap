// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches with snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root css-69i1ev"
    >
      <h2
        class="MuiBox-root css-qq7rji"
      >
        All Reviews:
      </h2>
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorWarning MuiIconButton-sizeLarge css-1mhl10j-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
    <div>
      <div
        class="MuiBox-root css-7bhehk"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1xy95es-MuiTypography-root"
        >
          comment1
        </p>
        <div
          class="MuiBox-root css-hbfgti"
        >
          user1 - Invalid Date
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1javkat-MuiDivider-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
